src/backend/expungeservice/test/test_expunger.py

```diff
@@ -20,16 +20,51 @@ def test_statute():
 
 def get_disp():
     date = datetime.date(1996, 1, 1)
-    disp_type = DispositionType.NO_CONVICTION
+    disp_type = DispositionType.CONVICTED
     return Disposition(disp_type, date)
 
+def get_qualifying_charge():
+    # Fix me with correct charge
+    disp = get_disp()
+    statute = Statute(113, 45, 5, 'd')
+    level = CrimeLevel('Infraction', 'A')
+    return Charge('Qualifiying Charge', statute, level,
+                  datetime.date(1995, 1, 1), disp)
+
+def get_unqualifying_charge():
+    # Fix me with correct charge
+    disp = get_disp()
+    statute = Statute(113, 45, 5, 'd')
+    level = CrimeLevel('Felony', 'A')
+    return Charge('Unqualifying Charge', statute, level,
+                  datetime.date(1995, 1, 1), disp)
+
 def test_expunger_classes():
     disp = get_disp()
     statute = Statute(113, 45, 5, 'd')
-    level = Level('Felony', 'A')
+    level = CrimeLevel('Felony', 'A')
     charges = [
-        Charge('Charge Name', statute, level,
-               datetime.date(1995, 1, 1), disp)
+        get_qualifying_charge(),
+        get_unqualifying_charge(),
     ]
-    cases = [Case(charges, CaseState.OPEN), 100.50]
+    cases = [
+        Case(charges, CaseState.OPEN, 100.50),
+        Case(charges, CaseState.CLOSED, 100.50),
+    ]
+    client = Client('John Doe', datetime.date(1970, 1, 1), cases)
+    assert(client.num_charges() == 4)
+
+def test_expunger():
+    charges = [
+        get_qualifying_charge(),
+        get_unqualifying_charge(),
+    ]
+    cases = [Case(charges, CaseState.OPEN, 100.50)]
     client = Client('John Doe', datetime.date(1970, 1, 1), cases)
+    record_analyzer = RecordAnalyzer(client)
+    time_elig_result = record_analyzer.analyze()
+
+    # TODO add individual test cases that test different paths in the decision tree
+    assert(charges[0].type_elig_result.code == ResultCode.FURTHER_ANALYSIS)
+    assert(charges[1].type_elig_result.code == ResultCode.INELIGIBLE)
+    assert(time_elig_result.code == ResultCode.OPEN_CASE)
```

Comment:
If the case is open it shouldn't stop us from analyzing the expungeability of a charge if the charge has a disposition associated to it.

Reply:
Yes. This is just testing if the time eligibility analyzer returns the result of "Open Case" instead of running its analysis (as per the flowchart). We need to add other tests to test expunge-ability of a single charge.

Reply:
Ah I see! Thanks